### ISSUE TITLE
How can i disable the sockjs.js module?

### ISSUE THREAD

**Issue by luneice** [webpack/webpack-dev-server#1430]
* Operating System: ubuntu 17.10
* Node Version: >=6.0.0
* NPM Version: >= 3.0.0
* webpack Version: 3.6.0
* webpack-dev-server Version: 2.9.1

i don't want it reports infomation, because it always gets errors and always sends some data which i think insignificance, how can i disable sockjs.js module?

**Comment by Viscerally** [comment]
Me too

**Comment by omarcresp** [comment]
it's a dependency for the hot reload? or we can disable sockjs from ours bundle???

**Comment by evilebottnawi** [comment]
If you disable `sockjs.js`, `hot` doesn't work

**Comment by zeroarst** [comment]
I think it is fine to disable sockjs.js on dev server and prod server. Because we don't need hot swap for both. I am seeing a lot of 404 errors as it is pointing to localhost:8080.

**Comment by michael-ciniawsky** [comment]
You currently can't and we are migrating away from that module asap, disabling will not really fix the problem :). @zeroarst Could you post the 404 here please? I will try and see what I can do about it in the meantime if there is anything commonly failing because of `sockjs` atm

**Comment by zeroarst** [comment]
I fixed it by disabling it. added `--inline=false` to my dev and prod scripts in package.json.

**Comment by neilyoung** [comment]
Did not work for me :(

**Comment by m0o066** [comment]
`--inline=false` not working for me either :(

**Comment by tomknappramos** [comment]
TLDR: 
`devServer.injectClient = false`

As far as I know, webpack inserts additional javascript into the client build to allow things like live reload and hot module replacement. And, I think that additional javascript uses sockjs-client to check the dev server for code changes (to then perform page reloads).

But, if you prevent webpack from inserting that additional javascript, then technically that also disables/prevents sockjs-client from getting placed into your dev bundle in the first place.

So, if you really want to disable the webpack client, one way to is to set,
`devServer.injectClient = false`

https://webpack.js.org/configuration/dev-server/#devserverinjectclient

**Comment by enessoylu** [comment]
If you are using react-app-rewired, that's the thing worked for me:

```js
// config-overrides.js
module.exports = function override(config, env) {
  // disable for client
  config.entry = config.entry.filter(
    (entry) => !entry.includes('webpackHotDevClient'),
  );

 // disable for server
  config.devServer = {
    hot: false,
    inline: false,
    liveReload: false,
    injectClient: false,
  };

 // ... rest of your config
}
```

**Comment by crosspath** [comment]
Solution for Rails-based projects:

```js
// file: config/webpack/environment.js

environment.config.devServer.injectClient = false
```

**Comment by feliarana** [comment]
Actually, you wanna put that in development.js, otherwise it will fail when you deploy to production

> Solution for Rails-based projects:
> 
> ```js
> // file: config/webpack/environment.js
> 
> environment.config.devServer.injectClient = false
> ```

**Comment by fevernova90** [comment]
With webpack v5, disabling websocket entirely (for whatever reason)
```javascript
// webpack.config.js
devServer: {
   client: false,
   webSocketServer: false
}
```

**Comment by alexander-akait** [comment]
`webSocketServer: false` should be enouth

**Comment by fevernova90** [comment]
> `webSocketServer: false` should be enough

Agreed.
But without `client: false`, I noticed that the served js still trying to connect to ws server, but returning 404 as server was disabled by `webSocketServer: false`. This may bother someone with OCD. lol

![image](https://user-images.githubusercontent.com/34376497/134512944-1842b88e-a929-4cb2-9465-a87164af355c.png)

**Comment by alexander-akait** [comment]
@fevernova90 I think it is bug
